sympy/solvers/tests/test_diophantine.py

```diff
@@ -583,6 +582,58 @@ def check_solutions(eq):
 
     return okay
 
+
+def check_linear(eq, param=symbols("t", Integer=True)):
```

Comment:
This function seems a bit complex for a test file. Would it be better to put it in the diophantine module itself?